GetIntoTeachingApi/Controllers/GetIntoTeaching/TeachingEventsController.cs

```diff
@@ -78,9 +78,20 @@ public class TeachingEventsController : ControllerBase
 
             var teachingEvents = await _store.SearchTeachingEventsAsync(request);
 
-            _metrics.TeachingEventSearchResults
-                .WithLabels(request.TypeId.ToString(), request.Radius.ToString())
-                .Observe(teachingEvents.Count());
+            // Temporary condition, to be removed when clients are updated and TypeId is removed
+            if (request.TypeIds != null && request.TypeId == null)
```

Comment:
I think this and the updates to the store could be simplified by instead updating the request model so that you override the `TypeIds` getter to either return its value (if set) or `new int[] { TypeId }` if null and `TypeId` is set. That way you don't need to have two sets of conditionals in the controller/store to account for the different methods and you can just use `request.TypeIds` everywhere.